packages/frontend/src/scripts/forced-actions/escape/finalizeEscapeForm.ts

```diff
@@ -0,0 +1,43 @@
+import { UserDetails } from '@explorer/shared'
+
+import { FINALIZE_ESCAPE_FORM_ID } from '../../../view'
+import { FinalizeEscapeFormProps } from '../../../view/pages/user/components/FinalizeEscapeForm'
+import { Wallet } from '../../peripherals/wallet'
+import { makeQuery } from '../../utils/query'
+
+export function initFinalizeEscapeForm() {
+  const { $ } = makeQuery(document.body)
+
+  const form = $.maybe(`#${FINALIZE_ESCAPE_FORM_ID}`)
+  if (!form) {
+    return
+  }
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+  const propsJson = JSON.parse(form.dataset.props ?? '{}')
+  const props = FinalizeEscapeFormProps.parse(propsJson)
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+  const userJson = JSON.parse(form.dataset.user ?? '{}')
+  const user = UserDetails.parse(userJson)
+  form.addEventListener('submit', (e) => {
+    e.preventDefault()
+    submitFinalizeEscape(props, user).catch(console.error)
+  })
+}
+
+async function submitFinalizeEscape(
+  props: FinalizeEscapeFormProps,
+  user: UserDetails
+) {
+  // // const hash = await Wallet.sendVerifyEscapeTransaction(
+  await Wallet.sendFinalizeEscapeTransaction(
+    user.address,
+    props.ownerStarkKey,
+    props.positionOrVaultId,
+    props.amount,
+    props.exchangeAddress
+  )
+
+  // // TODO: should we save via the API to our DB?
```

Comment:
Hmm, thats something to decide

Reply:
Good point, I knew about it but forgot to create a story. Here it is: L2B-1978